### PR TITLE
feat: Implement heartbeat support, docs and test

### LINKED ISSUE #2728: Support the heartbeat
@roman-kashitsyn [leaked on the forum](https://forum.dfinity.org/t/ic-cron-lets-schedule-some-tasks-bois/6506/6?u=nomeata) that the `canister_heartbeat` entry point is now supported by application subnets. The Interface spec doesn’t list it yet, but once it’s there we should probably support it from Motoko.

Unless we take the stance (from the discussion on the feature in the `ic-ref` at https://github.com/dfinity-lab/ic-ref/issues/245, which I can’t read any more) that `canister_heartbeat` is a low-level feature meant to be used mainly by canisters providing cron services to other canisters, and that Motoko canisters should not use this (inefficient, inflexible) interface, but should rather use dedicated cron-as-a-service-canisters.

But the pragmatic thing is probably to expose the `canister_heartbeat` entry point.

### PR DESCRIPTION
**Pull request by crusso**
Branch heartbeat #2971 for faster ci with access to cache.

### CODE REVIEW

File: test/run-drun/heartbeat.mo [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     };
+     Prim.debugPrint(debug_show {count});
+  };
+};
+
+//SKIP run
+//SKIP run-low
+//SKIP run-ir
+
+//CALL ingress go "DIDL\x00\x00"
```
Comment (crusso) [inline test/run-drun/heartbeat.mo:36]:
```suggestion
//CALL ingress go "DIDL\x00\x00"

```

File: doc/modules/language-guide/pages/language-manual.adoc [hunk 2]
```diff
@@ -1269,15 +1269,21 @@ The declaration `<dec>` of a `system` field must be a manifest `func` declaratio
 |===
 |  name | type | description
 
+| `heartbeat`  | `+() -> async ()+` | hearbeat action
```
Comment (ggreif) [inline doc/modules/language-guide/pages/language-manual.adoc:1272]:
```suggestion
| `heartbeat`  | `+() -> async ()+` | heartbeat action
```

**Review by ggreif — APPROVED** [review]
LGTM!

**Comment by ggreif** [comment]
Closes #2971.

**Comment by jzxchiang1** [comment]
🎉🎉

**Pull request merged**